registry/registry.go

```diff
@@ -38,24 +40,24 @@ type Registry struct {
  * This passes http.DefaultTransport to WrapTransport when creating the
  * http.Client.
  */
-func New(registryUrl, username, password string) (*Registry, error) {
+func New(ctx context.Context, registryUrl, username, password string) (*Registry, error) {
```

Comment:
Note that we'll  need to adjust the other consumer of this (the terraform plugin) when you update the vendor in amsterdam.  (Also I should rewrite the terraform plugin to use the new terraform "data provider" feature, or just get rid of it if we're not going to have any docker images in the config any more)